iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/grammar/DockerLexicalGrammar.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.docker.parser.grammar;
+
+import com.sonar.sslr.api.GenericTokenType;
+import java.util.Arrays;
+import org.sonar.iac.common.parser.grammar.LexicalConstant;
+import org.sonar.iac.common.parser.grammar.Punctuator;
+import org.sonar.sslr.grammar.GrammarRuleKey;
+import org.sonar.sslr.grammar.LexerlessGrammarBuilder;
+
+public enum DockerLexicalGrammar implements GrammarRuleKey {
+
+  FILE,
+  INSTRUCTION,
+
+  /**
+   * Lexical
+   */
+  EOF,
+
+  /**
+   * SPACING
+   */
+  SPACING,
+
+  /**
+   * INSTRUCTIONS
+   */
+  FROM;
+
+  public static LexerlessGrammarBuilder createGrammarBuilder() {
+    LexerlessGrammarBuilder b = LexerlessGrammarBuilder.create();
+
+    lexical(b);
+    punctuators(b);
+    keywords(b);
+
+    return b;
+  }
+
+  private static void punctuators(LexerlessGrammarBuilder b) {
+    for (Punctuator p : Punctuator.values()) {
+      b.rule(p).is(SPACING, p.getValue()).skip();
+    }
+  }
+  private static void lexical(LexerlessGrammarBuilder b) {
+    b.rule(SPACING).is(
+      b.skippedTrivia(b.regexp("[" + LexicalConstant.LINE_TERMINATOR + LexicalConstant.WHITESPACE + "]*+")),
+      b.zeroOrMore(
+        b.commentTrivia(b.regexp(LexicalConstant.COMMENT)),
```

Comment:
As mentioned in my previous comment, by doing so here we are parsing comment has either #, // or /* */, while in docker only # is supported.

Reply:
Good catch, changed to `DockerLexicalConstant.COMMENT`